File: app/src/main/java/denis/easyweather/app/net/WeatherService.java
package denis.easyweather.app.net;

import denis.easyweather.app.model.CityModel;
import denis.easyweather.app.model.ForecastModel;
import retrofit2.adapter.rxjava.Result;
import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

public interface WeatherService {

    @GET("/data/2.5/weather")
    Observable<CityModel> getWeatherByCityname(@Query("q") String q,
                                               @Query("units") String units,
                                               @Query("appid") String apiKey);

    @GET("/data/2.5/weather")
    Observable<Result<CityModel>> getWeatherByCoord(@Query("lat") String latitude,
                                                    @Query("lon") String longitude,
                                                    @Query("appid") String apiKey);

    @GET("data/2.5/forecast")
    Observable<Result<ForecastModel>> getWeatherForecastByCityname(@Query("q") String cityCountryName,
                                                                   @Query("appid") String apiKey);

}

File: app/src/main/java/denis/easyweather/app/net/WeatherService.kt
package denis.easyweather.app.net

import denis.easyweather.app.model.CityModel
import denis.easyweather.app.model.ForecastModel
import io.reactivex.Observable
import retrofit2.adapter.rxjava2.Result
import retrofit2.http.GET
import retrofit2.http.Query

interface WeatherService {

    @GET("/data/2.5/weather")
    fun getWeatherByCityname(@Query("q") q: String,
                             @Query("units") units: String,
                             @Query("appid") apiKey: String): Observable<CityModel>

    @GET("/data/2.5/weather")
    fun getWeatherByCoord(@Query("lat") latitude: String,
                          @Query("lon") longitude: String,
                          @Query("appid") apiKey: String): Observable<Result<CityModel>>

    @GET("data/2.5/forecast")
    fun getWeatherForecastByCityname(@Query("q") cityCountryName: String,
                                     @Query("appid") apiKey: String): Observable<Result<ForecastModel>>

}
